pkg/operator/ceph/controller/spec.go

```diff
@@ -662,16 +681,36 @@ func PrivilegedContext(runAsRoot bool) *v1.SecurityContext {
 }
 
 // LogCollectorContainer runs a cron job to rotate logs
-func LogCollectorContainer(daemonID, ns string, c cephv1.ClusterSpec) *v1.Container {
-	return &v1.Container{
+func LogCollectorContainer(daemonID, ns string, c cephv1.ClusterSpec) (*v1.Container, error) {
+	re := regexp.MustCompile("[0-9]")
+
+	periodicityInHour := re.FindString(c.LogCollector.Periodicity)
+	if c.LogCollector.Periodicity == "" && periodicityInHour == "" {
+		// default periodicity
+		periodicityInHour = "24"
+	} else {
+		// validate Periodicity
+		if strings.Contains(c.LogCollector.Periodicity, "h") || strings.Contains(c.LogCollector.Periodicity, "H") {
+			//  since this is already in hour no need to convert
+		} else if strings.Contains(c.LogCollector.Periodicity, "d") || strings.Contains(c.LogCollector.Periodicity, "D") {
+			intPeriodicity, _ := strconv.Atoi(periodicityInHour)
+			periodicityInHour = strconv.Itoa(intPeriodicity * 24)
+		} else {
+			logger.Error("not a valid periodicity, it should be in 2h/2d or 2H/2D.  Moving forward with defalult of 24h")
+			periodicityInHour = "24"
+		}
+	}
+
+	logSize := c.LogCollector.MaxLogSize.String()
```

Comment:
The format for MaxLogSize.String() may not always be what logrotate expects, so how about always converting to MB? Something like this:
```suggestion
	logSize := string.Sprintf("%dM", c.LogCollector.MaxLogSize.AsInt64() / 1024 / 1024)
```

Reply:
we are restricting with input that we take so no need to convertt now